Tools/TestWebKitAPI/Tests/WebCore/CSSTokenizerTest.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+
+#include <WebCore/CSSParserToken.h>
+#include <WebCore/CSSParserTokenRange.h>
+#include <WebCore/CSSTokenizer.h>
+#include <wtf/text/MakeString.h>
+#include <wtf/text/StringView.h>
+#include <wtf/text/WTFString.h>
+
+namespace TestWebKitAPI {
+
+using namespace WebCore;
+
+TEST(CSSTokenizerTest, TokenizeWhitespaceLikeInput)
+{
+    CSSTokenizer tokenizer("foo           \n\n\n bar"_s);
+    auto TokenRange = tokenizer.tokenRange();
+    auto TokenRangeSpan = TokenRange.span();
+
+    EXPECT_EQ(TokenRangeSpan.size(), 7ul);
```

Comment:
These are all backwards. Expected values come first in this macro. If the two values don’t match, the macro logs the value of the second one, and that should be the thing we are checking.

Also, it’s not good to code in the assumption that size_t is unsigned long. So this should either be `7U` if the width of the integer doesn’t matter or `(size_t)7` or a named constant of type `size_t`.